### PR TITLE
String.Substring helper

### PR DESCRIPTION
**Pull request by Andras-Csanyi**
This PR brings the following changes:
- Substring examples to the ConsoleApp
- Substring helper
- Substring tests

Notes:
- I made the changes on Mac, so .NET Framework tests and .NET Core 3.1 tests were not run, however I see that the CI build is executed on Windows Server. We will see if I introduced any suprise or not
- to avoid code duplication I treated the `end` parameter as `default` and it introduces a quirky corner case below
  - `Substring "foobar" 0 0` will result `foobar` due to `default(int) == 0` so, the code thinks `end` parameter were not be provided. From usage point of view I don't know when I would  need a `Substring "foobar" 0 0`. I'm open for suggestions. The case is provided by a test case.
- there is another interesting corner case `Substring "" 0` works, `string.Substring()` allows it. This is the reason only `null` input is checked by Guard and empty has an additional check.
- tests are provided for the exception paths too

Question:
- I see that test coverage can be increased by addtional test cases covering exception paths, do you see value in these? I'm happy to provide them as my time allows.
- I see that the the line endings are mixed LF and CRLF, it is intentional?
- Shall I extend the documentation with the new feature once it is approved and merged?

I'm looking forward to reading your feedback!

### CODE REVIEW

File: src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs [hunk 1]
```diff
@@ -412,6 +412,24 @@ public bool IsNotNullOrEmpty(string value)
         return !string.IsNullOrEmpty(value);
     }
 
+    [HandlebarsWriter(WriterType.String)]
+    public string Substring(string value, int start, int end = default)
```
Comment (StefH) [inline src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs:416]: Instead of using `int end = default` can you try to use `int? end = null` ?
(There are some other examples in my code what uses nullable as optional parameter.)

File: src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs [hunk 2]
```diff
@@ -412,6 +412,24 @@ public bool IsNotNullOrEmpty(string value)
         return !string.IsNullOrEmpty(value);
     }
 
+    [HandlebarsWriter(WriterType.String)]
+    public string Substring(string value, int start, int end = default)
+    {
+        Guard.NotNull(value);
+
+        if (value == string.Empty && start > 0)
```
Comment (StefH) [inline src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs:420]: Maybe you can just skip this check and let `value.Substring` just check + throw exception.

File: src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs [hunk 3]
```diff
@@ -412,6 +412,24 @@ public bool IsNotNullOrEmpty(string value)
...
+    public string Substring(string value, int start, int end = default)
+    {
+        Guard.NotNull(value);
+
+        if (value == string.Empty && start > 0)
+        {
+            throw new ArgumentException("When input string is empty start value cannot be higher than 0");
+        }
+
+        if (end == default)
```
Comment (StefH) [inline src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs:425]: Can be something like:
``` c#
return end is null ? value.Substring(start) : value.Substring(start, end.Value);
```

**Review by StefH — CHANGES_REQUESTED** [review]
Thanks for the PR.

Can you take a look at my comments.

**Comment by StefH** [comment]
CI Build is fine.

**Comment by StefH** [comment]
1. I see that test coverage can be increased by addtional test cases covering exception paths, do you see value in these? I'm happy to provide them as my time allows.
2. I see that the the line endings are mixed LF and CRLF, it is intentional?
3. Shall I extend the documentation with the new feature once it is approved and merged?


1--> Not really required.

2-->Not intentional, just an error I think...

3-->I'm not sure you can update the WIKI directly.
But you can copy-paste the updates maybe here?

File: src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs (hunk 1 above)
Comment (Andras-Csanyi) [reply src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs:416]: Done

File: src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs (hunk 2 above)
Comment (Andras-Csanyi) [reply src/Handlebars.Net.Helpers/Helpers/StringHelpers.cs:420]: Yes, makes more sense.

**Review by StefH — APPROVED** [review]
